library/random/formatter/alphanumeric_spec.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../../../spec_helper'
+
+ruby_version_is "3.1" do
+  require 'random/formatter'
+
+  describe "Random::Formatter#alphanumeric" do
+    before :each do
+      @object = Object.new
+      @object.extend(Random::Formatter)
+      @object.define_singleton_method(:bytes) do |n|
+        "\x00".b * n
+      end
```

Comment:
**suggestion:**

```ruby
def @object.bytes(n)
  "\x00".b * n
end
```

Reply:
Also it seems the `#bytes` method is a contract/interface between the `Random::Formatter` module and a class that includes it. So it might make sense to state it explicitly with a dedicated test case.

Reply:
Regarding random generation in these specs. `"\x00".b * n` seems to simple and is lacking randomness. So specs may not detect an issue with implementation.

With current `#bytes` method version:
```ruby
@object.alphanumeric # => "AAAAAAAAAAAAAAAA"
@object.alphanumeric(chars: ['a', 'b']) # => "aaaaaaaaaaaaaaaa"
```

So I would consider something more random:

```ruby
def @object.bytes(n)
  (0..255).to_a.shuffle.take(n).map { |b| b.chr }.join.b
end
```